src/dendrogram.jl

```diff
@@ -17,22 +18,39 @@ function treepositions(hc::Hclust, useheight::Bool)
         xs[:, i] .= [x1, x1, x2, x2]
         ys[:, i] .= [y1, ypos, ypos, y2]
     end
-
-    return xs, ys
+    if orientation == :h
+        return ys, xs
+    else
+        return xs, ys
+    end
 end
 
-@recipe function f(hc::Hclust; useheight=true)
+@recipe function f(hc::Hclust; useheight=true, orientation=:v )
     typeof(useheight) <: Bool || error("'useheight' argument must be true or false")
 
     legend := false
-    xforeground_color_axis := :white
-    xgrid := false
-    xlims := (0.5, length(hc.order) + 0.5)
-
     linecolor --> :black
-    xticks --> (1:nnodes(hc), (1:nnodes(hc))[hc.order])
-    ylims --> (0, Inf)
-    yshowaxis --> useheight
 
-    treepositions(hc, useheight)
+    if orientation == :h
+        if useheight
+            hs=sum(hc.heights)
+            xlims --> (0, hs+hs*0.01)
+        else
+            xlims --> (0, Inf)
+        end
+        xshowaxis --> useheight
+        yforeground_color_axis := :white
```

Comment:
IMO all `:=`s should be `-->`s in this recipe or is there an apparent reason that this should be non-overridable?

Reply:
Done as you said. Except for legend, as it was in the original code and it seems to make sense to disallow legends here.

Reply:
I'm fine with providing no legend by default, but I think if a user wants to have a legend, there should be a way to get one

Reply:
Of course. I have no opinion here, it is what the original author decided. Anyways, I changed it.